domain/flag/state/state.go

```diff
@@ -30,66 +30,73 @@ func NewState(factory coredb.TxnRunnerFactory, logger logger.Logger) *State {
 
 // SetFlag sets the value of a flag.
 // Description is used to describe the flag and it's potential state.
-func (s *State) SetFlag(ctx context.Context, flag string, value bool, description string) error {
+func (s *State) SetFlag(ctx context.Context, flagName string, value bool, description string) error {
 	db, err := s.DB()
 	if err != nil {
 		return errors.Trace(err)
 	}
 
-	query := `
-INSERT INTO flag (name, value, description)
-VALUES (?, ?, ?)
+	f := dbFlag{Name: flagName, Value: value, Description: description}
+
+	stmt, err := s.Prepare(`
+INSERT INTO   flag (name, value, description)
+VALUES        ($dbFlag.*)
 ON CONFLICT (name) DO UPDATE SET value = excluded.value,
                                  description = excluded.description;
-`
+`, f)
+	if err != nil {
+		return errors.Annotate(err, "preparing set flag stmt")
+	}
 
-	err = db.StdTxn(ctx, func(ctx context.Context, tx *sql.Tx) error {
-		result, err := tx.ExecContext(ctx, query, flag, value, description)
+	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		var outcome sqlair.Outcome
+		err := tx.Query(ctx, stmt, f).Get(&outcome)
 		if err != nil {
 			return errors.Trace(err)
 		}
-		if affected, err := result.RowsAffected(); err != nil {
+		if affected, err := outcome.Result().RowsAffected(); err != nil {
 			return errors.Trace(err)
 		} else if affected != 1 {
 			return errors.Errorf("unexpected number of rows affected: %d", affected)
```

Comment:
**suggestion(not-in-changelog):** In such error messages, having an indication of how much rows where expected to be affected can be handy.

```go
return errors.Errorf("unexpected number of rows affected: %d (should be 1)", affected)
```

Reply:
Nice, will add that